tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs

```diff
@@ -318,6 +318,20 @@ public void TiffDecoder_CanDecode_YCbCr_24Bit<TPixel>(TestImageProvider<TPixel>
             image.CompareToReferenceOutput(ImageComparer.Exact, provider);
         }
 
+        [Theory]
+        [WithFile(CieLab, PixelTypes.Rgba32)]
+        [WithFile(CieLabPlanar, PixelTypes.Rgba32)]
+        [WithFile(CieLabLzwPredictor, PixelTypes.Rgba32)]
+        public void TiffDecoder_CanDecode_CieLab<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            // Note: The image from MagickReferenceDecoder does not look right, maybe we are doing something wrong
+            // converting the pixel data from Magick.NET to our format with CieLab?
```

Comment:
What's happening here?

Reply:
The image `Tiff\CieLab.tiff` looks wrong when using the `MagickReferenceDecoder`:
![output](https://user-images.githubusercontent.com/38701097/179454882-89ec5709-5e59-4617-970d-4fcff83e6a39.png)

It should look like this:
![expected](https://user-images.githubusercontent.com/38701097/179454903-54dadb41-8981-4cac-ac3e-ad6610d478cb.png)

ImageMagick itself can decode this image just fine, that's why I thought we maybe doing something wrong in `MagickReferenceDecoder`.

Reply:
Ah right.... Yep, 

`IMagickImage<T>` contains `ColorSpace` enum which we are blindly ignoring when converting. I always assumed they did the same as us and converted color spaces on decode.

It looks like we should be checking that first and converting on the fly. 

@dlemstra Am I correct here?